pkg/provisioner/ironic/ironic.go

```diff
@@ -276,6 +276,51 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 				return result, errors.Wrap(err, "failed to create port in ironic")
 			}
 		}
+
+		if p.host.Spec.Image.URL != "" {
+			// FIXME(dhellmann): The Stein version of Ironic supports passing
+			// a URL. When we upgrade, we can stop doing this work ourself.
+			checksum, err := p.getImageChecksum()
+			if err != nil {
+				return result, errors.Wrap(err, "failed to retrieve image checksum")
+			}
+
+			p.log.Info("setting instance info",
+				"image_source", p.host.Spec.Image.URL,
+				"checksum", checksum,
+			)
+
+			updates := nodes.UpdateOpts{
+				nodes.UpdateOperation{
+					Op:    nodes.AddOp,
+					Path:  "/instance_info/image_source",
+					Value: p.host.Spec.Image.URL,
+				},
+				nodes.UpdateOperation{
+					Op:    nodes.AddOp,
+					Path:  "/instance_info/image_checksum",
+					Value: checksum,
+				},
+				// NOTE(dhellmann): We must fill in *some* value so that
+				// Ironic will monitor the host. We don't have a nova
+				// instance at all, so just give the node it's UUID again.
+				nodes.UpdateOperation{
+					Op:    nodes.ReplaceOp,
+					Path:  "/instance_uuid",
+					Value: p.host.Status.Provisioning.ID,
+				},
+			}
+			_, err = nodes.Update(p.client, ironicNode.UUID, updates).Extract()
+			switch err.(type) {
+			case nil:
+			case gophercloud.ErrDefault409:
+				p.log.Info("could not update host settings in ironic, busy")
+				result.Dirty = true
+				return result, nil
+			default:
+				return result, errors.Wrap(err, "failed to update host settings in ironic")
+			}
+		}
```

Comment:
I thought about refactoring that, but decided being explicit here was better. We always know the Op values and can just build a simple array. Go developers seem to be a lot more casual about repeating things. :shrug:

Reply:
Now that we're using this method to also recover from the ironic container being rescheduled, not calling `getUpdateOptsForNode()` means that Ironic will have less information after a resschedule than it did when the host was initially provisioned. Presumably this is not a problem, but I just want to leave a note because it could theoretically be a cause of inconsistencies in Ironic behaviour before and after a reschedule.